utils/src/main/kotlin/ArchiveUtils.kt

```diff
@@ -37,51 +39,52 @@ import org.apache.commons.compress.archivers.zip.ZipArchiveOutputStream
 import org.apache.commons.compress.compressors.bzip2.BZip2CompressorInputStream
 import org.apache.commons.compress.compressors.gzip.GzipCompressorInputStream
 
-private val UNCOMPRESSED_EXTENSIONS = listOf(".pom")
-private val TAR_EXTENSIONS = listOf(".gem", ".tar")
-private val TAR_BZIP2_EXTENSIONS = listOf(".tar.bz2", ".tbz2")
-private val TAR_GZIP_EXTENSIONS = listOf(".crate", ".tar.gz", ".tgz")
-private val ZIP_EXTENSIONS = listOf(".aar", ".egg", ".jar", ".war", ".whl", ".zip")
-private val SEVENZIP_EXTENSIONS = listOf(".7z")
-
-val ARCHIVE_EXTENSIONS = TAR_EXTENSIONS + TAR_BZIP2_EXTENSIONS + TAR_GZIP_EXTENSIONS + ZIP_EXTENSIONS +
-        SEVENZIP_EXTENSIONS
+enum class ArchiveType(vararg val extensions: String) {
+    TAR(".gem", ".tar"),
+    TAR_BZIP2(".tar.bz2", ".tbz2"),
+    TAR_GZIP(".crate", ".tar.gz", ".tgz"),
+    ZIP(".aar", ".egg", ".jar", ".war", ".whl", ".zip"),
+    SEVENZIP(".7z"),
+    POM(".pom"), // Special case of a "fake archive".
+    NONE(""); // This must come last as the empty string matches anything.
+
+    companion object {
+        fun getType(filename: String): ArchiveType {
+            val lowerName = filename.toLowerCase()
+            return enumValues<ArchiveType>().find { type ->
+                type.extensions.any { lowerName.endsWith(it) }
```

Comment:
How aboutinstead : `type != NONE && type.extensions.any { lowerName.endsWith(it) }` ?

This eliminates any assumptions about the order the enum elements are defined and thus is less error prone. It also removes the need for the code comment next to NONE.

Reply:
Done, but yet slightly differently to avoid multiple checks in the inner loop.